.github/workflows/release.yaml

```diff
@@ -0,0 +1,56 @@
+name: Release Packages
+
+on:
+  push:
+    branches:
+      - main
+      - 'prerelease/*'
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release to NPM
+    runs-on: ['self-hosted', 'org', 'npm-publish']
+    permissions:
+      contents: write
+      id-token: write
+      pull-requests: write
+      repository-projects: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Akeyless Get Secrets
+        id: get_auth_token
+        uses:
+          docker://us-west1-docker.pkg.dev/devopsre/akeyless-public/akeyless-action:latest
+        with:
+          api-url: https://api.gateway.akeyless.celo-networks-dev.org
+          access-id: p-kf9vjzruht6l
+          static-secrets: '{"/static-secrets/apps-tooling-circle/npm-publish-token":"NPM_TOKEN"}'
+      - name: Setup Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: 'Setup yarn'
+        shell: bash
+        run: |
+          npm install --global yarn
+          source ~/.bashrc
+      - name: Install Dependencies
+        shell: bash
+        run: yarn
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ env.NPM_TOKEN }}
+        with:
+          # This expects you to have a script called release which does a build for your packages and calls changeset publish
+          publish: yarn release
+      - name: Remove Yarn
+        # https://github.com/actions/setup-node/issues/182
+        if: always()
+        run: npm uninstall -g yarn
```

Comment:
The runner lifecycle is the job lifecycle (created and destroyed for each job), so probably we can skip removing anything?

Reply:
oh According the issue above for self hosted runners it isnt destroyed? 

but i dont know